### PR TITLE
[SAFE] computation of range and azimuth resolution

### PR DESCRIPTION
**Pull request by johntruckenbrodt**

### CODE REVIEW

**Comment by johntruckenbrodt** [comment]
@avalentino may I ask you to have a brief look at this? For a GRD this returns
```
(25.09280788069985, 21.361175923321102)
```
which seems to high. For an SLC this is
```
(3.107103655284058, 21.646363075695035)
```

**Comment by avalentino** [comment]
@johntruckenbrodt the computation is correct if you consider 5 totally independent looks.
For the S1-GRD I think that some look overlap is used changing a little git the computation.

For S1 IW GRD-HR the ENL is 4.4 (see [[1]](https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/document-library/-/asset_publisher/1dO7RF5fJMbd/content/sentinel-1-product-definition)).

If you use 4.4 instead of 5 in your computation you should have a resolution that is approximatively 22m.

I don't remember if and how this information can be retrieved form metadata.
Will try to provide more information ASAP.

**Comment by johntruckenbrodt** [comment]
Thanks @avalentino, I think this explains it. Wouldn't it then be easiest to just compute the number of looks as `processingBandwidth / lookBandwidth`? This results in almost exactly 4 looks and 20 m resolution. If we would incorporate the `lookOverlap` we would get:
`processingBandwidth + lookOverlap * (numberOfLooks - 1) / lookBandwidth = 5`

So basically the image is multi-looked to a size equivalent to four independent looks by using five looks with some overlap.
`lookOverlap / lookBandwidth = 0.25`

File: pyroSAR/drivers.py [hunk 1]
```diff
@@ -1758,62 +1758,50 @@ def resolution(self):
...
+            wrg = float(rg_proc.find('windowCoefficient').text)
+            brg = float(rg_proc.find('processingBandwidth').text)
+            lbrg = float(rg_proc.find('lookBandwidth').text)
+            lrg = brg / lbrg
+            kbrg = b_factors[coefficients.index(wrg)]
+            resolutions_rg.append(0.886 * c / (2 * brg) * kbrg * lrg)
+            
+            # computation of azimuth resolution; yet to be checked for correctness
+            az_proc = par.find('azimuthProcessing')
+            laz = int(az_proc.find('numberOfLooks').text)
```
Comment (avalentino) [inline pyroSAR/drivers.py:1791]: Probably it makes sense to use the same formula used for RG, it should work also for GRD MR and LR.

Comment (johntruckenbrodt) [reply pyroSAR/drivers.py]: done

**Comment by johntruckenbrodt** [comment]
@avalentino, thanks again. You only refer to GRD. Isn't this computation also correct for SLCs? This is what I get now:
SLC: `(3.107103655284058, 21.646363075695035)`
GRD: `(20.03277058905459, 21.361175923321102)`

**Comment by avalentino** [comment]
> You only refer to GRD. Isn't this computation also correct for SLCs?

Yes you are right, for SLC you should always find that the number of looks is 1 but the formula indeed is the same

**Comment by johntruckenbrodt** [comment]
Certainly. Computing the looks for SLC is not necessary but the result is still 1.0 and it saves a few lines of code. Indeed, the division by the sine of the incidence angle is done but only for GRDs. I think we got it now. I will merge this once I am ready to release the next version.

**Pull request merged**